scripts/sync_repos.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash -eux
+#
+# Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+# Exceptions. See /LICENSE for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# Sync directories in the main Carbon repository into dedicated child
+# repositories to better match repository-oriented installing and tooling.
+
+ORIGIN_DIR="$PWD"
+COMMIT_SHA="$(git rev-parse --short $GITHUB_SHA)"
+COMMIT_SUMMARY="Original $(git show -s --pretty=full "$COMMIT_SHA")
+
+$(git diff --summary "${COMMIT_SHA}^!")
+"
```

Comment:
This implementation feels pretty difficult to understand right now, and I think the multiline strings are adding to my difficult. What would you think of using Python with an eye on long-term maintainability?